### PR TITLE
General code cleanup, improved naming, etc.

### PR DESCRIPTION
**Pull request by thinkyhead**
- Rename vars for `BARICUDA` and `FILRUNOUT` features
- Rename `setTargetedHotend` to `set_target_extruder_for_mcode`
- Rename `Stop()` to `stop()`
- Rename `min_pos`/`max_pos` to `sw_endstop_min`/`sw_endstop_max`

### CODE REVIEW

**Comment by jbrazio** [comment]
> Rename setTargetedHotend to set_target_extruder_for_mcode

@thinkyhead I find the `_for_mcode` part too much, if we add this suffix to this then we should change some others. I would remove it.

File: ArduinoAddons/Arduino_1.0.x/libraries/U8glib/utility/u8g_rot.c [hunk 1]
```diff
@@ -40,7 +40,11 @@ uint8_t u8g_dev_rot90_fn(u8g_t *u8g, u8g_dev_t *dev, uint8_t msg, void *arg);
...
 uint8_t u8g_dev_rot270_fn(u8g_t *u8g, u8g_dev_t *dev, uint8_t msg, void *arg);
 
+#ifdef __XC8
 uint8_t u8g_dev_rot_dummy_fn(void *u8g, void *dev, uint8_t msg, void *arg)
+#else
+uint8_t u8g_dev_rot_dummy_fn(u8g_t *u8g, u8g_dev_t *dev, uint8_t msg, void *arg)
+#endif
 {
   return 0;
 }
```
Comment (Blue-Marlin) [inline ArduinoAddons/Arduino_1.0.x/libraries/U8glib/utility/u8g_rot.c]: Don't patch u8glib!
Better copy in the new version.

> 2016-04-05 v1.19   
> - Fixed contrast (brightness) setting for SH1106 and SSD1306 devices
> - Fixed compiler warnings

Comment (jbrazio) [inline ArduinoAddons/Arduino_1.0.x/libraries/U8glib/utility/u8g_rot.c]: Actually.. who is using this ?
`ArduinoAddons/Arduino_1.6.x/hardware/marlin/avr/libraries/U8glib/utility/u8g_rot.c`

See #3468

**Comment by thinkyhead** [comment]
> I find the `_for_mcode` part too much

@jbrazio

Well I felt that the function should be distinguished as being a utility function specifically for use by some GCode handler functions. Also the name was too general to carry any meaning, so it seemed important to note (in the name) that the argument it takes is a GCode command code (as opposed to an index, an ID number, or something that you would normally associate with "getting a temperature.")

So, given that this is a utility function used only by the GCode handlers, what kind of naming convention would you recommend?

**Comment by thinkyhead** [comment]
> Don't patch u8glib! Better copy in the new version.

I would be happy to do so. It makes sense that they already patched this small glitch.

How has the project decided (historically) the version of **U8glib** (and other libs) to bundle with Marlin? Does it matter if we call the bundled version "certified" or give a "guarantee" that it is compatible?

**Comment by jbrazio** [comment]
@thinkyhead 

> what kind of naming convention would you recommend?

`protected void func();` ? jk :-P

I said "I would remove it" but we could otherwise apply the same suffix for all the other `func()` related with the g-code parser, in the future this will actually do our lives a bit easier due to the grouping.

**Comment by thinkyhead** [comment]
I will mull it over and update this PR soon. I feel that the function's name should be expressive of what it does and clarify its relationship to the parser family of functions. (In the long run, it will end up being a member function in the `Parser` class and have a nicer name.) Perhaps I will move it up near to `code_seen` and the other GCode helper functions.

**Comment by thinkyhead** [comment]
I ended up making the function name even longer (`get_target_extruder_from_command`). But at least it's grouped with the other GCode helpers now.

**Comment by AnHardt** [comment]
> How has the project decided (historically) the version of U8glib (and other libs) to bundle with Marlin? Does it matter if we call the bundled version "certified" or give a "guarantee" that it is compatible?

Historically seen there was a initial commit with v1.11 and an update by me to the then actual v1.17.
The history of that update is, that i always worked with the newest version and one day realized we had a bundled one. I just had switched from Arduino 1.0.x to 1.5.x so i knew it would work on the relevant Ardoino versions.

Since some time i'm now using Arduino 1.6.x and do not have any problems with u8glib v1.19, so an update seems to be possible. But there are no relevant changes in there, what could be important for  us - so an update is not necessary now.

If compiled with all warnings on, even the newest u8glib throws a huge amount of warnings. Patching only one ore two of them makes no sense. a.) We'd loose the ability to throw in a random new u8glib because we'd be afraid to lose our work (f.e. to support a new hardware).  And more important b.) It would keep you away from improving Marlin.

I'd guess with Arduino 1.6.x it's so convenient to get u8glib via the IDE, that using our bundled version will be the rare exception now, or at least in the near future. 
If we have to drop the support for Arduino 1.0.x one day, that could be the opportunity to get rid of the bundled u8glib. If than, one day, there is a version of u8glib, not working well with Marlin, we can throw in the "last known good" version again.

There are no known version conflicts or incompatibility's between u8glib v1.11 - v1.19 and ArduinoIDE 1.0.3 up to 1.6.8 (that is only the range i looked for).

**Pull request merged**